In [6]:
import pandas as pd
import numpy as np
import os

##### Carregando os dados

In [7]:
arquivo_csv = os.path.join("adult.csv")

df = pd.read_csv(arquivo_csv)

print(df.head())

   age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country incom

##### Limpando valores ausentes

In [8]:
import numpy as np

df["occupation"] = df["occupation"].replace("?", np.nan)
df = df.dropna(subset=["occupation"])

##### Questão 1: número de obsevações, descrição das variáveis, identificação das variáveis dependentes e das explicativas

In [9]:
print(f"Número de observações: {len(df)}")

Número de observações: 46033


In [10]:
df = pd.read_csv(arquivo_csv)
df.head()
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype
---  ------           --------------  -----
 0   age              48842 non-null  int64
 1   workclass        48842 non-null  str  
 2   fnlwgt           48842 non-null  int64
 3   education        48842 non-null  str  
 4   educational-num  48842 non-null  int64
 5   marital-status   48842 non-null  str  
 6   occupation       48842 non-null  str  
 7   relationship     48842 non-null  str  
 8   race             48842 non-null  str  
 9   gender           48842 non-null  str  
 10  capital-gain     48842 non-null  int64
 11  capital-loss     48842 non-null  int64
 12  hours-per-week   48842 non-null  int64
 13  native-country   48842 non-null  str  
 14  income           48842 non-null  str  
dtypes: int64(6), str(9)
memory usage: 9.3 MB



| Variável          | Tipo        | Descrição                                                                                                  | Identificação   |
| ----------------- | ----------- | ---------------------------------------------------------------------------------------------------------- | --------------- |
| `income`          | Binária     | Indica a faixa de renda anual do indivíduo, sendo **0 = renda ≤ US$ 50.000** e **1 = renda > US$ 50.000**. | **Dependente**  |
| `age`             | Numérica    | Idade do indivíduo, medida em anos.                                                                        | **Explicativa** |
| `educational-num` | Numérica    | Número de anos de escolaridade do indivíduo.                                                               | **Explicativa** |
| `hours-per-week`  | Numérica    | Número de horas trabalhadas por semana.                                                                    | **Explicativa** |
| `gender`          | Qualitativa | Sexo do indivíduo (Masculino ou Feminino).                                                                 | **Explicativa** |
| `occupation`      | Qualitativa | Ocupação ou profissão exercida pelo indivíduo.                                                             | **Explicativa** |


* d. justificativa para a escolha da vari´avel dependente bin´aria.


##### 2º Questão: 

In [11]:
df.describe()

,age,fnlwgt,educational-num,capital-gain,capital-loss,hours-per-week
count,48842.000000,4.884200e+04,48842.000000,48842.000000,48842.000000,48842.000000
mean,38.643585,1.896641e+05,10.078089,1079.067626,87.502314,40.422382
std,13.710510,1.056040e+05,2.570973,7452.019058,403.004552,12.391444
min,17.000000,1.228500e+04,1.000000,0.000000,0.000000,1.000000
25%,28.000000,1.175505e+05,9.000000,0.000000,0.000000,40.000000
50%,37.000000,1.781445e+05,10.000000,0.000000,0.000000,40.000000
75%,48.000000,2.376420e+05,12.000000,0.000000,0.000000,45.000000
max,90.000000,1.490400e+06,16.000000,99999.000000,4356.000000,99.000000


In [12]:
#Frequência da variável income(variável dependente)
freq_income = df["income"].value_counts()

print(freq_income)

income
<=50K    37155
>50K     11687
Name: count, dtype: int64


A variável dependente income apresenta um desbalanceamento entre as classes. Dos 48.842 registros da base de dados, 37.155 indivíduos (aproximadamente 76,1%) pertencem à classe 0, enquanto 11.687 (23,9%) pertencem à classe 1. Esse desbalanceamento deve ser considerado na etapa de modelagem, pois pode influenciar o desempenho dos algoritmos de classificação.

In [13]:
#Frequência da variável gender
print(df["gender"].value_counts())


gender
Male      32650
Female    16192
Name: count, dtype: int64


A variável gender mostra predominância do sexo masculino, com 32.650 registros (66,8%), enquanto o sexo feminino representa 16.192 registros (33,2%). Essa diferença indica uma distribuição desigual entre os gêneros na base analisada.

In [14]:
# Frequência da variável age
print(df["age"].value_counts())

age
36    1348
35    1337
33    1335
23    1329
31    1325
      ... 
88       6
85       5
87       3
89       2
86       1
Name: count, Length: 74, dtype: int64


A variável age possui 74 valores distintos, variando aproximadamente de 17 a 90 anos. As idades mais frequentes concentram-se entre 30 e 40 anos, destacando-se os indivíduos de 36 anos (1.348 registros), 35 anos (1.337), 33 anos (1.335), 23 anos (1.329) e 31 anos (1.325). Nas idades mais elevadas observa-se uma frequência bastante reduzida, indicando menor representatividade de pessoas acima de 80 anos.

In [15]:
#Frequência da variável educational-num
print(df["educational-num"].value_counts())

educational-num
9     15784
10    10878
13     8025
14     2657
11     2061
7      1812
12     1601
6      1389
4       955
15      834
5       756
8       657
16      594
3       509
2       247
1        83
Name: count, dtype: int64


A variável educational-num, que representa o nível educacional em formato numérico, apresenta maior concentração nos valores 9 (15.784 registros), 10 (10.878 registros) e 13 (8.025 registros). Os níveis educacionais extremos possuem frequências significativamente menores, indicando que a maior parte da população concentra-se em níveis intermediários de escolaridade.

In [16]:
#Frequência da variável hours-per-week
print(df["hours-per-week"].value_counts())

hours-per-week
40    22803
50     4246
45     2717
60     2177
35     1937
      ...  
79        1
69        1
87        1
94        1
82        1
Name: count, Length: 96, dtype: int64


A variável hours-per-week indica que a carga horária semanal mais comum é de 40 horas, com 22.803 registros, representando quase metade da amostra. Também são frequentes jornadas de 50, 45, 60 e 35 horas semanais, enquanto cargas horárias muito baixas ou muito elevadas aparecem com baixa frequência.

In [17]:
#Frequência da variável occupation
print(df["occupation"].value_counts())

occupation
Prof-specialty       6172
Craft-repair         6112
Exec-managerial      6086
Adm-clerical         5611
Sales                5504
Other-service        4923
Machine-op-inspct    3022
?                    2809
Transport-moving     2355
Handlers-cleaners    2072
Farming-fishing      1490
Tech-support         1446
Protective-serv       983
Priv-house-serv       242
Armed-Forces           15
Name: count, dtype: int64


Na variável occupation, as ocupações mais frequentes são Prof-specialty (6.172 registros), Craft-repair (6.112), Exec-managerial (6.086), Adm-clerical (5.611) e Sales (5.504). Observa-se ainda a presença de 2.809 registros com o valor "?", indicando dados ausentes ou não informados. Antes da construção de modelos preditivos, recomenda-se tratar esses registros, seja substituindo-os por valores apropriados, seja considerando-os como uma categoria específica, conforme a estratégia de pré-processamento adotada.

##### Letra D)

##### Questão 3: Explique, do ponto de vista teórico:

* o que são variáveis dummies;
    - Variáveis dummies (ou variáveis binárias) são variáveis qualitativas representadas numericamente pelos valores 0 e 1, indicando, respectivamente, a ausência ou a presença de uma determinada característica. Elas permitem incorporar informações categóricas em modelos econométricos, como sexo, estado civil, escolaridade, região geográfica ou setor de atividade.

* qual sua finalidade em modelos econométricos; 
    - A principal finalidade das variáveis dummies é permitir que características qualitativas influenciem a variável dependente em um modelo econométrico. Elas possibilitam estimar o efeito de atributos que não podem ser medidos numericamente, como gênero, região, ocupação ou nível de escolaridade.
    Em modelos de regressão, os coeficientes associados às variáveis dummies representam a diferença média entre um grupo específico e a categoria de referência, mantendo constantes as demais variáveis explicativas. Dessa forma, é possível avaliar, por exemplo, se indivíduos com ensino superior apresentam renda diferente daqueles sem ensino superior, controlando outros fatores do modelo.

* o significado da categoria de referência;
    - A categoria de referência (ou categoria-base) é o grupo utilizado como comparação para interpretar os coeficientes das variáveis dummies. Essa categoria não recebe uma variável dummy própria quando o modelo possui intercepto, evitando problemas de multicolinearidade.
     - Por exemplo, considerando a variável gênero:
        - Mulher = 1
        - Homem = 0

    - Os homens constituem a categoria de referência. Assim, o coeficiente estimado para a dummy "Mulher" representa a diferença média entre mulheres e homens, mantendo constantes as demais variáveis do modelo.

* o problema da armadilha das variáveis dummies (Dummy Variable Trap);
   - A Dummy Variable Trap ocorre quando todas as categorias de uma variável qualitativa são incluídas no modelo juntamente com o intercepto. Nessa situação, uma dummy pode ser escrita como combinação linear das demais, gerando multicolinearidade perfeita, o que impede a estimação dos coeficientes pelo método dos Mínimos Quadrados Ordinários (MQO).
    Por exemplo, para representar quatro regiões (Norte, Nordeste, Sul e Sudeste), não se deve incluir quatro variáveis dummies juntamente com o intercepto, pois a soma das quatro será sempre igual a 1. A solução consiste em: 
        - manter o intercepto e utilizar apenas k − 1 variáveis dummies, deixando uma categoria como referência; ou incluir todas as k dummies e retirar o intercepto do modelo.
        - Na prática, a abordagem mais utilizada é manter o intercepto e omitir uma das categorias, permitindo que todos os coeficientes sejam interpretados em relação à categoria de referência.

##### Questão 4:  Caso necessário, construa as variáveis dummies da base de dados.


* A)
- foi realizada a limpeza da variável income, removendo possíveis espaços em branco:
df["income"] = df["income"].str.strip()
- variável foi transformada em uma dummy, atribuindo:

    - 0 para indivíduos com renda anual menor ou igual a US$ 50.000 (<=50K);
    - 1 para indivíduos com renda anual superior a US$ 50.000 (>50K).

- O procedimento foi realizado por meio do comando:
    df["income"] = df["income"].map({
    "<=50K": 0,
    ">50K": 1
})

* B)A categoria de referência foi:
    - Income = 0, correspondente aos indivíduos com renda anual menor ou igual a US$ 50.000.


* C) A variável dummy income identifica se um indivíduo pertence ao grupo de maior renda. O valor 0 representa indivíduos com renda anual de até US$ 50.000 (categoria de referência), enquanto o valor 1 representa indivíduos com renda superior a US$ 50.000.

In [18]:
#removendo espaços em branco no início e no final da coluna "income"
df["income"] = df["income"].str.strip()

#transformando a coluna "income" em valores binários (0 e 1)(variável dummy)
df["income"] = df["income"].map({
    "<=50K": 0,
    ">50K": 1
})
print(df["income"].unique())

[0 1]


In [19]:
#transformanodo a variável "gender" em valores binários
# Dummy para gênero
    # 0 = Feminino (categoria de referência) e 1 = Masculino

df["male"] = df["gender"].map({
    "Female": 0,
    "Male": 1
})

print(df[["gender", "male"]].head())



   gender  male
0    Male     1
1    Male     1
2    Male     1
3    Male     1
4  Female     0


##### A coluna occupation também é uma dummy , mas vou estimala com stats model , quando estimar o modelo

##### Questão 5: Estime inicialmente um Modelo de Probabilidade Linear (MQO)

In [20]:
#a. equação do modelo de regressão linear múltipla (MPL) para a variável dependente "income" e as variáveis independentes "age", "educational-num", "hours-per-week", "gender" e "occupation".
import statsmodels.formula.api as smf

modelo_mpl = smf.ols(
    'income ~ age + Q("educational-num") + Q("hours-per-week") + C(gender) + C(occupation)',
    data=df
).fit()

print(modelo_mpl.summary())
modelo_mpl.summary()

                            OLS Regression Results                            
Dep. Variable:                 income   R-squared:                       0.240
Model:                            OLS   Adj. R-squared:                  0.240
Method:                 Least Squares   F-statistic:                     855.5
Date:                qui, 06 ago 2026   Prob (F-statistic):               0.00
Time:                        17:02:16   Log-Likelihood:                -21005.
No. Observations:               48842   AIC:                         4.205e+04
Df Residuals:                   48823   BIC:                         4.222e+04
Df Model:                          18                                         
Covariance Type:            nonrobust                                         
                                         coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------------------
Inte

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                 income   R-squared:                       0.240
Model:                            OLS   Adj. R-squared:                  0.240
Method:                 Least Squares   F-statistic:                     855.5
Date:                qui, 06 ago 2026   Prob (F-statistic):               0.00
Time:                        17:02:16   Log-Likelihood:                -21005.
No. Observations:               48842   AIC:                         4.205e+04
Df Residuals:                   48823   BIC:                         4.222e+04
Df Model:                          18                                         
Covariance Type:            nonrobust                                         
======================================================================================================
                                         coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------------------
Intercept                             -0.6704      0.012    -56.157      0.000      -0.694      -0.647
C(gender)[T.Male]                      0.1566      0.004     38.934      0.000       0.149       0.164
C(occupation)[T.Adm-clerical]          0.0388      0.009      4.444      0.000       0.022       0.056
C(occupation)[T.Armed-Forces]          0.1269      0.096      1.317      0.188      -0.062       0.316
C(occupation)[T.Craft-repair]          0.0389      0.009      4.452      0.000       0.022       0.056
C(occupation)[T.Exec-managerial]       0.2150      0.009     24.331      0.000       0.198       0.232
C(occupation)[T.Farming-fishing]      -0.0801      0.012     -6.561      0.000      -0.104      -0.056
C(occupation)[T.Handlers-cleaners]    -0.0363      0.011     -3.311      0.001      -0.058      -0.015
C(occupation)[T.Machine-op-inspct]     0.0033      0.010      0.336      0.737      -0.016       0.023
C(occupation)[T.Other-service]        -0.0054      0.009     -0.612      0.541      -0.023       0.012
C(occupation)[T.Priv-house-serv]       0.0307      0.025      1.228      0.220      -0.018       0.080
C(occupation)[T.Prof-specialty]        0.1682      0.009     18.636      0.000       0.151       0.186
C(occupation)[T.Protective-serv]       0.0993      0.014      7.127      0.000       0.072       0.127
C(occupation)[T.Sales]                 0.0988      0.009     11.274      0.000       0.082       0.116
C(occupation)[T.Tech-support]          0.1095      0.012      8.998      0.000       0.086       0.133
C(occupation)[T.Transport-moving]      0.0111      0.011      1.041      0.298      -0.010       0.032
age                                    0.0056      0.000     44.689      0.000       0.005       0.006
Q("educational-num")                   0.0356      0.001     44.844      0.000       0.034       0.037
Q("hours-per-week")                    0.0040      0.000     27.147      0.000       0.004       0.004
==============================================================================
Omnibus:                     4049.499   Durbin-Watson:                   2.012
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             4883.549
Skew:                           0.757   Prob(JB):                         0.00
Kurtosis:                       2.670   Cond. No.                     3.34e+03
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 3.34e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

##### b. interpretação econômica dos coeficientes:
* No Modelo de Probabilidade Linear, os coeficientes representam a variação na probabilidade de um indivíduo possuir renda anual superior a US$ 50.000, mantendo as demais variáveis constantes.

    * Intercepto (-0,6704): representa a probabilidade estimada para um indivíduo pertencente às categorias de referência e com as variáveis numéricas iguais a zero. Como esse valor é negativo, ele não possui interpretação econômica direta, evidenciando uma limitação do Modelo de Probabilidade Linear.
    * Male (0,1566): indivíduos do sexo masculino possuem uma probabilidade aproximadamente 15,66 pontos percentuais maior de apresentar renda superior a US$ 50.000 em relação às mulheres, mantendo as demais variáveis constantes.
    * Age (0,0056): cada ano adicional de idade aumenta em aproximadamente 0,56 ponto percentual a probabilidade de um indivíduo possuir renda superior a US$ 50.000.
    * Educational-num (0,0356): cada ano adicional de escolaridade aumenta em aproximadamente 3,56 pontos percentuais a probabilidade de o indivíduo possuir renda superior a US$ 50.000.
    * Hours-per-week (0,0040): cada hora adicional trabalhada por semana aumenta em aproximadamente 0,40 ponto percentual a probabilidade de o indivíduo possuir renda superior a US$ 50.000.
    * Occupation: os coeficientes das ocupações representam a diferença na probabilidade de obter renda superior a US$ 50.000 em relação à categoria de referência da variável occupation. Por exemplo:
    * Exec-managerial (0,2150): indivíduos em cargos gerenciais apresentam probabilidade 21,50 pontos percentuais maior de possuir renda superior a US$ 50.000.
    * Prof-specialty (0,1682): aumenta essa probabilidade em 16,82 pontos percentuais.
    * Farming-fishing (-0,0801): reduz essa probabilidade em 8,01 pontos percentuais.
    * As ocupações Armed-Forces, Machine-op-inspct, Other-service, Priv-house-serv e Transport-moving não apresentaram coeficientes estatisticamente significativos (valor-p > 0,05), indicando que não há evidências de diferença em relação à categoria de referência.

##### c. coeficiente de determinação R2:
* R-squared: 0.240 e Adj. R-squared: 0.240
*O modelo apresentou R² = 0,240 e R² ajustado = 0,240. Isso significa que aproximadamente 24% da variação da probabilidade de um indivíduo possuir renda superior a US$ 50.000 é explicada pelas variáveis explicativas incluídas no modelo.

#### d. limitações do modelo:


* Segundo Gujarati, o Modelo de Probabilidade Linear apresenta algumas limitações importantes:

    * As probabilidades estimadas podem assumir valores menores que 0 ou maiores que 1, o que não faz sentido do ponto de vista probabilístico.
    * Os erros tendem a apresentar heterocedasticidade, violando uma das hipóteses do Modelo Clássico de Regressão Linear.
    * O efeito marginal das variáveis explicativas é constante, independentemente do nível da probabilidade, o que pode não representar adequadamente o comportamento dos dados.

##### Questão 6: Explique teoricamente os modelos Logit e Probit. Responda:

* a. Por que o MQO não é o método mais adequado para variável dependente binária?
    - O Modelo de Probabilidade Linear (MQO) apresenta algumas limitações quando a variável dependente é binária:
        - pode gerar probabilidades menores que 0 ou maiores que 1;
        - os erros tendem a apresentar heterocedasticidade, violando uma hipótese do Modelo Clássico de Regressão Linear;
        - assume que o efeito das variáveis explicativas sobre a probabilidade é constante, o que nem sempre representa adequadamente a realidade.

* b. Diferenças entre os modelos Logit e Probit
    - A principal diferença entre os modelos está na distribuição utilizada para representar a probabilidade.
        - Modelo Logit: utiliza a distribuição logística para modelar a probabilidade do evento ocorrer.
        - Modelo Probit: utiliza a distribuição normal padrão (distribuição acumulada da normal).

* c. Vantagens e limitações de cada modelo
    - Modelo Logit
        - Vantagens:
            - garante probabilidades entre 0 e 1;
            - possui interpretação baseada na razão de chances (odds ratio);
            - é amplamente utilizado em estudos econômicos, financeiros e sociais.

        - Limitações:
            - os coeficientes não representam diretamente o efeito sobre a probabilidade, sendo necessária a análise dos efeitos marginais para uma interpretação mais intuitiva.
    - Modelo Probit
        - Vantagens:
            - garante probabilidades entre 0 e 1;
            - baseia-se na distribuição normal, adequada quando se assume que existe uma variável latente normalmente distribuída;
            - também produz estimativas consistentes para variáveis binárias.
        - Limitações:
            - os coeficientes também não possuem interpretação direta em termos de probabilidade;
            - interpretação costuma ser menos intuitiva do que a do modelo Logit.

##### Questão 7: Estime um modelo Logit.

In [21]:
import statsmodels.formula.api as smf

modelo_logit = smf.logit(
    'income ~ age + Q("educational-num") + Q("hours-per-week") + C(gender) + C(occupation)',
    data=df
).fit()

print(modelo_logit.summary())

Optimization terminated successfully.
         Current function value: 0.412483
         Iterations 9
                           Logit Regression Results                           
Dep. Variable:                 income   No. Observations:                48842
Model:                          Logit   Df Residuals:                    48823
Method:                           MLE   Df Model:                           18
Date:                qui, 06 ago 2026   Pseudo R-squ.:                  0.2504
Time:                        17:22:43   Log-Likelihood:                -20146.
converged:                       True   LL-Null:                       -26875.
Covariance Type:            nonrobust   LLR p-value:                     0.000
                                         coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------------------------------
Intercept                             -8.9397      0.125    

a. Equação estimada
    - ln(1−P(income=1) P(income=1))= 
            −8,9397+1,1695(Male)
            +0,7318(Adm-clerical)+1,3296(Armed-Forces)
            +0,8580(Craft-repair)+1,5419(Exec-managerial)
            −0,3653(Farming-fishing)−0,0810(Handlers-cleaners)
            +0,4946(Machine-op-inspct)−0,3918(Other-service)
            +⋯+0,0450(Age)
            +0,2697(Educational-num)+0,0322(Hours-per-week)

b. Interpretação dos sinais dos coeficientes

- No modelo Logit, o sinal do coeficiente indica se a variável aumenta ou reduz a probabilidade de um indivíduo possuir renda superior a US$ 50.000.

    - Coeficiente positivo: aumenta a probabilidade do evento ocorrer.
    - oeficiente negativo: reduz a probabilidade do evento ocorrer.

    - Assim, observa-se que:

        - Male possui coeficiente positivo, indicando maior probabilidade de renda superior a US$ 50.000 em comparação às mulheres.
        - Age, Educational-num e Hours-per-week apresentam coeficientes positivos, sugerindo que maior idade, maior escolaridade e mais horas trabalhadas aumentam a probabilidade de renda elevada.
        - As ocupações Exec-managerial, Prof-specialty, Craft-repair, Adm-clerical, Sales e Tech-support apresentam coeficientes positivos.
        - As ocupações Farming-fishing e Other-service apresentam coeficientes negativos, indicando menor probabilidade de renda superior a US$ 50.000 em relação à categoria de referência.

c. Significância estatística
- A significância estatística é avaliada pelo valor-p.
Observa-se que praticamente todas as variáveis apresentaram valor-p inferior a 0,05, indicando que são estatisticamente significativas ao nível de 5%.
- LLR p-value = 0,000 indica que o modelo Logit é estatisticamente significativo como um todo.


    - As exceções apresentadas na saída foram:

        - Handlers-cleaners (p = 0,488), que não é estatisticamente significativa;
        - (caso outras ocupações no trecho omitido apresentem p > 0,05, elas também não serão significativas).

d. Interpretação econômica dos resultados

- Os resultados indicam que características individuais e ocupacionais influenciam significativamente a probabilidade de um indivíduo possuir renda superior a US$ 50.000.

    - Ser do sexo masculino aumenta a probabilidade de renda elevada.
    - O aumento da idade está associado a maior probabilidade de renda superior a US$ 50.000.
    - Maior escolaridade exerce efeito positivo sobre essa probabilidade, indicando que indivíduos com mais anos de estudo tendem a obter maiores rendimentos.
    - O aumento das horas trabalhadas por semana também eleva a probabilidade de renda superior a US$ 50.000.
    - Entre as ocupações, cargos como Exec-managerial, Prof-specialty e Adm-clerical aumentam a probabilidade de renda elevada, enquanto ocupações como Farming-fishing e Other-service apresentam efeito negativo quando comparadas à categoria de referência.
    - Pseudo R² = 0,2504,as variáveis explicativas contribuem para explicar a probabilidade de o indivíduo possuir renda superior a US$ 50.000.

	​


##### Questão 8: Estime um modelo Probit

In [22]:
import statsmodels.formula.api as smf

modelo_probit = smf.probit(
    'income ~ age + Q("educational-num") + Q("hours-per-week") + C(gender) + C(occupation)',
    data=df
).fit()

print(modelo_probit.summary())

Optimization terminated successfully.
         Current function value: 0.412570
         Iterations 8
                          Probit Regression Results                           
Dep. Variable:                 income   No. Observations:                48842
Model:                         Probit   Df Residuals:                    48823
Method:                           MLE   Df Model:                           18
Date:                qui, 06 ago 2026   Pseudo R-squ.:                  0.2502
Time:                        17:31:59   Log-Likelihood:                -20151.
converged:                       True   LL-Null:                       -26875.
Covariance Type:            nonrobust   LLR p-value:                     0.000
                                         coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------------------------------
Intercept                             -5.0926      0.067    

a. Equação estimada
-A equação estimada do modelo Probit é:
    - Φ**−1(P(income=1))= −5,0926+0,6472(Male)+0,3854(Adm-clerical)+0,7184(Armed-Forces)+0,4460(Craft-repair)0,8656(Exec-managerial)−0,2160(Farming-fishing)−0,0533(Handlers-cleaners)+0,2475(Machine-op-inspct)−0,1679(Other-service)+⋯+0,0260(Age)+0,1548(Educational-num)+0,0183(Hours-per-week)

b. Interpretação dos coeficientes
- No modelo Probit, os coeficientes não representam diretamente a variação da probabilidade. Eles indicam como cada variável influencia um índice latente que determina a probabilidade de o indivíduo possuir renda superior a US$ 50.000.
    - Coeficiente positivo: aumenta a probabilidade de income = 1.
    - Coeficiente negativo: reduz a probabilidade de income = 1.

    - O coeficiente de Male (0,6472) é positivo, indicando que homens apresentam maior probabilidade de possuir renda superior a US$ 50.000 quando comparados às mulheres.
    - A variável Age (0,0260) possui efeito positivo, indicando que indivíduos mais velhos tendem a apresentar maior probabilidade de renda elevada.
    - Educational-num (0,1548) apresenta coeficiente positivo, sugerindo que maior escolaridade aumenta a probabilidade de renda superior a US$ 50.000.
    - Hours-per-week (0,0183) também apresenta efeito positivo, indicando que mais horas trabalhadas por semana aumentam essa probabilidade.
    - Entre as ocupações, Exec-managerial, Craft-repair, Adm-clerical e Machine-op-inspct possuem efeito positivo, enquanto Farming-fishing e Other-service apresentam efeito negativo em relação à categoria de referência.

c. Comparação com o modelo Logit
- Os resultados do modelo Probit foram bastante semelhantes aos obtidos pelo modelo Logit.
- Ambos apresentaram os mesmos sinais para os coeficientes, indicando a mesma direção dos efeitos das variáveis explicativas.
- As variáveis idade, escolaridade, horas trabalhadas e gênero permaneceram estatisticamente significativas nos dois modelos.
- As ocupações Exec-managerial, Craft-repair e Adm-clerical apresentaram efeito positivo e significativo nos dois modelos, enquanto Farming-fishing e Other-service apresentaram efeito negativo.
- Algumas categorias, como Handlers-cleaners, não foram estatisticamente significativas em ambos os modelos.
- A categoria Armed-Forces foi significativa no Logit (valor-p = 0,031), mas no Probit apresentou valor-p = 0,054, ficando ligeiramente acima do nível de significância de 5%. Assim, no Probit ela não é considerada estatisticamente significativa a 5%, embora seja significativa a 10%.


##### Questão 9: Compare os resultados obtidos pelos modelos MQO, Logit e Probit. Discuta

a. Diferenças nos coeficientes
    - No Modelo de Probabilidade Linear (MQO), os coeficientes representam a variação direta na probabilidade de um indivíduo possuir renda superior a US$ 50.000.
    - No Logit, os coeficientes representam alterações no logaritmo da razão de chances (log-odds).
    - No Probit, os coeficientes representam alterações em um índice baseado na distribuição normal padrão.
b. Diferenças na significância estatística
    - As variáveis idade, educational-num, hours-per-week e gender foram estatisticamente significativas em todos os modelos (valor-p < 0,05).
    - Entre as ocupações, a maioria também permaneceu significativa nos três modelos. Entretanto, ocorreram pequenas diferenças. Por exemplo:
    - A categoria Armed-Forces foi significativa no modelo Logit (valor-p = 0,031), mas não foi significativa no Probit ao nível de 5% (valor-p = 0,054).
    - A categoria Handlers-cleaners não apresentou significância estatística tanto no Logit quanto no Probit.

c. Vantagens e limitações de cada modelo
| Modelo                                   | Vantagens                                                                                                                                            | Limitações                                                                                                                                                  |
| ---------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------- |
| **MQO (Modelo de Probabilidade Linear)** | Simples de estimar e interpretar. Os coeficientes representam diretamente a variação na probabilidade.                                               | Pode prever probabilidades menores que 0 ou maiores que 1 e apresenta heterocedasticidade quando a variável dependente é binária.                           |
| **Logit**                                | Garante probabilidades entre 0 e 1, produz estimativas consistentes e permite interpretar os resultados por meio da razão de chances (*odds ratio*). | Os coeficientes não representam diretamente alterações na probabilidade, sendo necessário calcular efeitos marginais para uma interpretação mais intuitiva. |
| **Probit**                               | Também garante probabilidades entre 0 e 1 e utiliza a distribuição normal padrão, sendo adequado para variáveis binárias.                            | Assim como o Logit, seus coeficientes não representam diretamente alterações na probabilidade e exigem efeitos marginais para interpretação econômica.      |




##### Questão 10: Calcule os efeitos marginais do modelo Logit ou Probit.

a. Efeitos marginais médios
* Os efeitos marginais médios (Average Marginal Effects – AME) foram obtidos a partir do modelo Logit e representam a variação média na probabilidade de um indivíduo possuir renda superior a US$ 50.000 quando uma variável explicativa aumenta uma unidade, mantendo as demais constantes.

In [ ]:
# Efeitos marginais médios (Average Marginal Effects - AME)
efeitos = modelo_logit.get_margeff(at='overall')
print(efeitos.summary())

        Logit Marginal Effects       
Dep. Variable:                 income
Method:                          dydx
At:                           overall
                                        dy/dx    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------------------------------
C(gender)[T.Male]                      0.1551      0.004     37.382      0.000       0.147       0.163
C(occupation)[T.Adm-clerical]          0.0970      0.011      8.747      0.000       0.075       0.119
C(occupation)[T.Armed-Forces]          0.1763      0.082      2.159      0.031       0.016       0.336
C(occupation)[T.Craft-repair]          0.1138      0.011     10.786      0.000       0.093       0.134
C(occupation)[T.Exec-managerial]       0.2045      0.010     19.938      0.000       0.184       0.225
C(occupation)[T.Farming-fishing]      -0.0484      0.015     -3.201      0.001      -0.078      -0.019
C(occupation)[T.Handlers

b. Interpretação econômica: Os efeitos marginais indicam o impacto médio de cada variável sobre a probabilidade de um indivíduo possuir renda superior a US$ 50.000.

- Male (0,1551): ser do sexo masculino aumenta, em média, a probabilidade de possuir renda superior a US$ 50.000 em aproximadamente 15,5 pontos percentuais, em relação às mulheres.
- Age (0,0060): cada ano adicional de idade aumenta a probabilidade de renda superior a US$ 50.000 em cerca de 0,6 ponto percentual.
- Educational-num (0,0358): um ano adicional de escolaridade aumenta, em média, a probabilidade de renda superior a US$ 50.000 em aproximadamente 3,6 pontos percentuais.
- Hours-per-week (0,0043): cada hora adicional trabalhada por semana aumenta a probabilidade de renda superior a US$ 50.000 em cerca de 0,4 ponto percentual.

Em relação às ocupações:

- Exec-managerial (0,2045) apresentou o maior efeito positivo, aumentando a probabilidade em aproximadamente 20,5 pontos percentuais em relação à categoria de referência.
- Prof-specialty (0,1663) aumenta essa probabilidade em aproximadamente 16,6 pontos percentuais.
- Tech-support (0,1611) aumenta a probabilidade em cerca de 16,1 pontos percentuais.
- Sales (0,1314) aumenta a probabilidade em aproximadamente 13,1 pontos percentuais.
- Farming-fishing (-0,0484) reduz a probabilidade em cerca de 4,8 pontos percentuais.
- Other-service (-0,0520) reduz a probabilidade em aproximadamente 5,2 pontos percentuais.
- Priv-house-serv (-0,2052) apresentou o maior efeito negativo, reduzindo a probabilidade em aproximadamente 20,5 pontos percentuais.

c. Comparação entre coeficientes e efeitos marginais
- Os coeficientes do Logit representam alterações no logaritmo da razão de chances (log-odds) e, portanto, não indicam diretamente o impacto sobre a probabilidade.
- Os efeitos marginais convertem esses coeficientes em mudanças na probabilidade, tornando a interpretação econômica mais simples e intuitiva.

##### Questão 11: Avalie a capacidade preditiva do modelo.

##### Questão 12: Considere as seguintes afirmações. Classifique cada uma como verdadeira ou falsa e justifique.

##### Questão 13: Elabore um parecer técnico (máximo de uma página) resumindo os resultados encontrados. Seu texto deverá abordar: